### PR TITLE
Transformers prepare + Modularize attention

### PR DESCRIPTION
**Pull request by Eleven1Liu**
**Description**
1. Enable to load the pretrained language model from transformers.
2. Add new networks
    - BERTAttention:`bert_attention.py`
    - BiLSTMLWMHAN(BiLSTM + LabelWise MultiHead AtteNtion):  `labelwise_attention_networks.py`
3. Batch tokenize text when using the transformer's encoder.
4. Modularize `LabelwiseAttention` and `LabelwiseMultiHeadAttention`.

**Note**
Feature branch `modularize_attention` has been merged to this branch.
Please see the code difference between the two branches in PR #149.

### CODE REVIEW

File: libmultilabel/nn/data_utils.py [hunk 1]
```diff
@@ -23,21 +23,29 @@
...
         return len(self.data)
 
     def __getitem__(self, index):
         data = self.data[index]
+        input_ids = data['text']
+
+        if self.tokenizer is not None:
+            input_ids = self.tokenizer.encode(data['text'], add_special_tokens=False)
+        else:
+            data['text'] = tokenize(data['text'])
```
Comment (Eleven1Liu) [inline libmultilabel/nn/data_utils.py:45]: `default_tokenizer` ?

File: libmultilabel/nn/networks/bert_attn.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    """
+    def __init__(
+        self,
+        num_classes,
+        dropout=0.2,
+        embedding_dim=512,
+        lm_weight='bert-base-cased',
+        lm_window=512,
+        num_heads=2
+    ):
```
Comment (Eleven1Liu) [inline libmultilabel/nn/networks/bert_attn.py:27]:
```suggestion
**kwargs):
```

File: libmultilabel/nn/networks/bert_attn.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    def forward(self, input):
+        input_ids = input['text'] # (batch_size, sequence_length)
+        x = self.lm_feature(input_ids) # (batch_size, sequence_length, lm_hidden_size)
+        attention_mask = input_ids == self.lm.config.pad_token_id
+        x = self.embed_drop(x)
+        x = self.lm_linear(x)  # (batch_size, sequence_length, embedding_dim)
+
+        k = v = x.permute(1, 0, 2) # (sequence_length, batch_size, embedding_dim)
+        q = self.query.repeat(1, input_ids.size(0), 1) # (1, batch_size, embedding_dim)
```
Comment (henryyang42) [inline libmultilabel/nn/networks/bert_attn.py:85]: I found that current modification is not label-wise attention.
`q` here should follow caml to use random initialized ICD embedding, which has shape of `(label_size, batch_size, embedding_dim)`

Comment (henryyang42) [reply libmultilabel/nn/networks/bert_attn.py:85]:
```
self.icd_embed = nn.Parameter(torch.Tensor(args.class_num, args.lm_embedding_dim), requires_grad=True)
xavier_uniform_(self.icd_embed.data)
```
Another thing to take care of is that we also apply ``xavier_uniform_``  to `nn.Parameter`

File: example_config/MIMIC-50/bert_attn.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+# model
+model_name: BERTAttention
+init_weight: null
+network_config:
+  dropout: 0.2
+  lm_weight: emilyalsentzer/Bio_ClinicalBERT
+  lm_window: 512
+  embedding_dim: 512
+  num_heads: 2
```
Comment (henryyang42) [inline example_config/MIMIC-50/bert_attn.yml:33]: NOTE: ``num_heads: 8`` in production setting
